mt-kahypar/partition/refinement/flows/parallel_construction.cpp

```diff
@@ -229,37 +229,36 @@ FlowProblem ParallelConstruction<GraphAndGainTypes>::constructDefault(const Part
     vec<whfc::Node>& tmp_pins = _tmp_pins.local();
     for ( size_t i = start; i < end; ++i ) {
       const HyperedgeID he = sub_hg.hes[i];
-      if ( !FlowNetworkConstruction::dropHyperedge(phg, he, block_0, block_1) ) {
+      FlowNetworkEdgeParameters parameters =
+        FlowNetworkConstruction::getParameters(phg, _context, he, block_0, block_1);
+
+      if ( parameters.capacity > 0 ) {
```

Comment:
Maybe add a comment here and in sequential construction that this mechanism is also used to implement the filter which edges we keep. I found it a bit surprising.